### ISSUE TITLE
Expiration 

### ISSUE THREAD

**Issue by lollita** [FiloSottile/mkcert#343]
Do it is possible to set custom expiration date?

**Comment by ikeyan** [comment]
https://github.com/FiloSottile/mkcert/blob/0a3190b1659e514d6e9b03eedfa25049d046000b/cert.go#L59-L62
expiration period is hardcoded here, and the comment tells the reason of the value.

**Comment by ikeyan** [comment]
if this is a feature request, maybe duplicate of https://github.com/FiloSottile/mkcert/issues/339.

**Comment by andykais** [comment]
maybe expose this as an option and if a user exceeds the macos maximum expiration date, log a warning? Also, I'll be honest I am new to certificates, but I _think_ that a cert generated on linux or windows is not valid on mac, and vice versa because it is tied to a root cert that is specific to that OS. So if we detect what OS the cert is being generated for, we could avoid even logging that error for non-macs.

The big use case for me is distributing a localhost web app as a linux package. The certificate doesn't _ever_ need to expire in that case, and it would be frustrating for users to see an error in their app that their cert expired 2 years later

**Comment by prasanthbazz** [comment]
@andykais Certificates are platform/ OS neutral AFAIK. It is the way we install certificate to the OS certificate store differ between OS.

**Comment by rubencodes** [comment]
Follow-up question on this - is there any way to detect an expired certificate? So I can create + trust a new one?

**Comment by ronaaron** [comment]
I would definitely appreciate an option to set the certificate expiration in number of days. Thank you!

**Comment by ed2050** [comment]
>  // Certificates last for 2 years and 3 months, which is always less than 
>  // 825 days, the limit that macOS/iOS apply to all certificates, 
>  // including custom roots. See https://support.apple.com/en-us/HT210176. 
>  expiration := time.Now().AddDate(2, 3, 0) 
> 
> expiration period is hardcoded here, and the comment tells the reason of the value.

Actually [that link](https://support.apple.com/en-us/HT210176) only mentions server certificate expiry: 
"TLS *server certificates* must have a validity period of 825 days or fewer (as expressed in the NotBefore and NotAfter fields of the certificate)."

There's nothing about custom roots.  Where did you get that info from?

**Comment by ed2050** [comment]
> The big use case for me is distributing a localhost web app as a linux package. The certificate doesn't _ever_ need to expire in that case, and it would be frustrating for users to see an error in their app that their cert expired 2 years later

Did you ever solve this?  I'm in similar situation.  2 years is far too short for local root CA.

**Comment by redoper1** [comment]
> [mkcert/cert.go](https://github.com/FiloSottile/mkcert/blob/0a3190b1659e514d6e9b03eedfa25049d046000b/cert.go#L59-L62)
> 
> Lines 59 to 62 in [0a3190b](/FiloSottile/mkcert/commit/0a3190b1659e514d6e9b03eedfa25049d046000b)
> 
>  // Certificates last for 2 years and 3 months, which is always less than 
>  // 825 days, the limit that macOS/iOS apply to all certificates, 
>  // including custom roots. See https://support.apple.com/en-us/HT210176. 
>  expiration := time.Now().AddDate(2, 3, 0) 
> 
> expiration period is hardcoded here, and the comment tells the reason of the value.

Yeah, it's hardcoded, but it would be great if we could set shorter expire date in days for testing purposes by some flag. Default could be as it is.

**Comment by encryptblockr** [comment]
can we make this as part of parameters when creating certs??? hard coding expiry date seems like anti-software

some people want shorter duration and some want longer duration

just pass parameter like `--expiration` or `-expiration` so we can set it and that can take string value like `--expiration "5 years"` or `-expiration "100 years"`
